src/main/java/io/github/tryexceptelse/jdex/be/Contact.java

```diff
@@ -1,52 +1,191 @@
 package io.github.tryexceptelse.jdex.be;
 
-import io.github.tryexceptelse.jdex.be.entries.ContactEntry;
-import io.github.tryexceptelse.jdex.be.entries.ContactNotes;
+import io.github.tryexceptelse.jdex.be.entries.*;
 
 import java.io.Serializable;
 import java.util.HashMap;
 
 /**
- * One object of this class represents one contact to be added to the list
- * storing contacts.
+ * One object of this class represents one contact to be added to the list. Each field takes its own object
+ * and may have specialized methods.
  */
 public class Contact implements IContact, Serializable
 {
-    private HashMap<String, ContactEntry> entries; // HashMap storing Contact's entries.
+    private FirstName first;
+    private LastName last;
+    private EmailAddress email;
+    private StreetAddress streetAddr;
+    private PhoneNumber phone;
+    private ContactNotes notes;
 
     /**
      * Default constructor for Contact.
-     * Stores empty map of entries.
      */
-    public Contact(){
-        // placeholder
+    public Contact()
+    {
+
+    }
+
+    /**
+     * @param first:      expects FirstName object representing the first name of the contact
+     * @param last:       expect LastName object representing the last name of the contact
+     * @param email:      expects EmailAddress object representing the email address of the contact
+     * @param streetAddr: expects EmailAddress object represnting the email address of the contact
+     * @param phone:      expects PhoneNumber object representing the phone number of the contact
+     * @param notes:      expects ContactNotes object representing notes for the contact
+     */
+    public Contact(FirstName first,
+                   LastName last,
+                   EmailAddress email,
+                   StreetAddress streetAddr,
+                   PhoneNumber phone,
+                   ContactNotes notes)
+    {
+        this.first=first;
+        this.last=last;
+        this.email=email;
+        this.streetAddr=streetAddr;
+        this.phone=phone;
+        this.notes=notes;
+    }
+    /**
+     * Calls the checkHasValidString method on each of the parameters of the object
+     */
+    public boolean checkParametersAreValid()
+    {
+        // Checking to make sure all passed fields are valid. The methods are defined in their own classes
+        if (last.checkHasValidString())
```

Comment:
checkHasValidString returns true if the string is valid, and judging by its use elsewhere, this method is also meant to return true if all entries are valid. This method as written however returns false if any of the ContactEntry objects are valid.